lib/entities/team-membership.ts

```diff
@@ -1,38 +1,41 @@
-/**
- * TeamMembership instances
- * @namespace TeamMembership
- */
-
 import cloneDeep from 'lodash/cloneDeep'
 import { freezeSys, toPlainObject } from 'contentful-sdk-core'
 import enhanceWithMethods from '../enhance-with-methods'
 import errorHandler from '../error-handler'
+import { DefaultElements, MetaSysProps, QueryOptions, CollectionProp } from '../common-types'
+import { AxiosInstance } from 'axios'
 
-/**
- * @memberof TeamMembership
- * @typedef TeamMembership
- * @prop {Meta.Sys} sys - System metadata
- * @prop {boolean} admin - User is an admin
- * @prop {function(): Object} toPlainObject() - Returns this Team Membership as a plain JS object
- */
+export interface Options {
+  teamId?: string
+  query?: QueryOptions
+}
 
-/**
- * @memberof TeamMembership
- * @typedef TeamMembershipCollection
- * @prop {number} total
- * @prop {number} limit
- * @prop {number} skip
- * @prop {Object<{type: "Array"}>} sys
- * @prop {Array<TeamMembership.TeamMembership>} items
- * @prop {function(): Object} toPlainObject() - Returns the collection as a plain JS object
- */
+export type TeamMembershipProps = {
+  /**
+   * System metadata
+   */
+  sys: MetaSysProps
 
-function createTeamMembershipApi(http) {
+  /**
+   * Is admin
+   */
+  admin: boolean
+
+  /**
+   * Organization membership id
+   */
+  organizationMembershipId: string
+}
+
+export interface TeamMembership extends TeamMembershipProps, DefaultElements<TeamMembershipProps> {
+  delete(): Promise<void>
+  update(): Promise<TeamMembership>
+}
+
+function createTeamMembershipApi(http: AxiosInstance) {
   return {
     /**
```

Comment:
Could you move the comment from implementation to the type definition? Otherwise, they won't show up in an automatically generated documentation.